otelcollector/shared/configmap/ccp/tomlparser-ccp-default-targets-metrics-keep-list.go

```diff
@@ -15,11 +15,12 @@ var (
 	configSchemaVersion, minimalIngestionProfile                           string
 	controlplaneApiserverRegex, controlplaneClusterAutoscalerRegex         string
 	controlplaneKubeSchedulerRegex, controlplaneKubeControllerManagerRegex string
-	controlplaneEtcdRegex                                                  string
+	controlplaneNodeAutoProvisioningRegex, controlplaneEtcdRegex           string
 	controlplaneKubeControllerManagerMinMac                                = "rest_client_request_duration_seconds|rest_client_requests_total|workqueue_depth"
 	controlplaneKubeSchedulerMinMac                                        = "scheduler_pending_pods|scheduler_unschedulable_pods|scheduler_pod_scheduling_attempts|scheduler_queue_incoming_pods_total|scheduler_preemption_attempts_total|scheduler_preemption_victims|scheduler_scheduling_attempt_duration_seconds|scheduler_schedule_attempts_total|scheduler_pod_scheduling_duration_seconds"
 	controlplaneApiserverMinMac                                            = "apiserver_request_total|apiserver_cache_list_fetched_objects_total|apiserver_cache_list_returned_objects_total|apiserver_flowcontrol_demand_seats_average|apiserver_flowcontrol_current_limit_seats|apiserver_request_sli_duration_seconds_count|apiserver_request_sli_duration_seconds_sum|process_start_time_seconds|apiserver_request_duration_seconds_count|apiserver_request_duration_seconds_sum|apiserver_storage_list_fetched_objects_total|apiserver_storage_list_returned_objects_total|apiserver_current_inflight_requests"
 	controlplaneClusterAutoscalerMinMac                                    = "rest_client_requests_total|cluster_autoscaler_((last_activity|cluster_safe_to_autoscale|scale_down_in_cooldown|scaled_up_nodes_total|unneeded_nodes_count|unschedulable_pods_count|nodes_count))|cloudprovider_azure_api_request_(errors|duration_seconds_(bucket|count))"
+	controlplaneNodeAutoProvisioningMinMac                                 = "karpenter_((nodes_created_total|nodes_terminated_total|voluntary_disruption_eligible_nodes|nodeclaims_disrupted_total|voluntary_disruption_decisions_total|pods_state))"
```

Comment:
note rest_client_requests_total isn't included in this list, (per question here I think it would need to be if it is to be included in mipfalse-keepmetrics.yaml: https://github.com/Azure/prometheus-collector/pull/1169/files#r2124446547)